pkg/controller.v1beta1/trial/trial_controller_util.go

```diff
@@ -70,6 +70,14 @@ func (r *ReconcileTrial) UpdateTrialStatusCondition(instance *trialsv1beta1.Tria
 			msg := "Metrics are not available"
 			reason := TrialMetricsUnavailableReason
 
+			// If the type of metrics collector is Push, We should insert an unavailable value to Katib DB
+			if instance.Spec.MetricsCollector.Collector != nil &&
```

Comment:
We might need to add an integration test for it: https://github.com/kubeflow/katib/blob/4b477e82d56433389a70da2ffc54088ebc9c449d/pkg/controller.v1beta1/trial/trial_controller_test.go
cc @tenzen-y

Reply:
Additionally, please try to run the example with Push collector when you don't insert objective metrics and verify that Trial has been moved into `MetricsUnavailable` status.

Reply:
> We might need to add an integration test for it: https://github.com/kubeflow/katib/blob/4b477e82d56433389a70da2ffc54088ebc9c449d/pkg/controller.v1beta1/trial/trial_controller_test.go
cc @tenzen-y

@andreyvelich If I understand correctly, do you mean we need to add an integration test for Push Collector?

> Additionally, please try to run the example with Push collector when you don't insert objective metrics and verify that Trial has been moved into MetricsUnavailable status.

SGTM.

Reply:
@andreyvelich I tried to run the example with `report_metrics` commented. And the result shows that all the Trials have been moved into `MetricsUnavailable` status.

<img width="966" alt="截屏2024-08-02 16 36 35" src="https://github.com/user-attachments/assets/fc1ec4e3-db5e-476c-9d6c-a6d6b77a5902">

Reply:
>@andreyvelich If I understand correctly, do you mean we need to add an integration test for Push Collector?

Yes, just to verify that appropriate functional calls are happening.

Reply:
Okay, I'll implement that in the next few commits.

Reply:
The integration test for Push Collector has been added now. Since there are no differences for MCs in successful scenario, I only add a test case in which the batch job is completed but no metrics are reported to the Katib DB.

Does it meet our need? WDYT👀 @andreyvelich

Reply:
Yes, I think, it should be good. Thank you!